src/library/vicon_estimator.cpp

```diff
@@ -186,46 +214,64 @@ Eigen::Vector3d RotationalEstimator::getEstimatedRate() const {
   return rate_estimate_B_;
 }
 
-void RotationalEstimator::updateEstimate(
-    const Eigen::Quaterniond& orientation_measured_B_W) {
+EstimatorStatus RotationalEstimator::updateEstimate(
+    const Eigen::Quaterniond& orientation_measured_B_W,
+    const double timestamp) {
   // Writing the raw measurement to the intermediate results structure
   estimator_results_.orientation_measured_ = orientation_measured_B_W;
 
   // Writing the old estimate to the intermediate results structure
   estimator_results_.orientation_old_ = orientation_estimate_B_W_;
   estimator_results_.rate_old_ = rate_estimate_B_;
 
-  // Detecting outlier measurements
-  bool measurement_update_flag;
-  if (detectMeasurementOutlier(orientation_measured_B_W)) {
-    measurement_update_flag = false;
-  } else {
-    measurement_update_flag = true;
+  // Performing some initialization if this is the first measurement.
+  if (first_measurement_flag_) {
+    first_measurement_flag_ = false;
+    orientation_measured_old_ = orientation_measured_B_W;
+    last_timestamp_ = timestamp;
```

Comment:
I think this is a well spent additional code, since otherwise you get a biased estimate from the beginning on. Also it is pretty common to initialize a EKF with a good initial guess if available.